### ISSUE TITLE
BetterBags doesn't respect C_Container.SetItemSearch

### ISSUE THREAD

**Issue by Protuhj** [Cidan/BetterBags#584]
### Addon Version

v0.1.96

### World of Warcraft Version

Retail

### Describe the bug

Using the WoW API call `C_Container.SetItemSearch("search string")` BB doesn't filter.

This worked in AdiBags, and should work in BB too.
It also works with the default bag.

### Steps to reproduce

1. Open the bag
2. Run the command in chat: `/run C_Container.SetItemSearch("soulbound")` soulbound items should be highlighted.

**Comment by Cidan** [comment]
Thanks for the report.

Since we have our own comprehensive search engine, there is no way that I know of to get the parameter passed into SetItemSearch cleanly. The Blizzard event does not return the actual search term, unfortunately. If I discover a way to do this, I'll get it done, but it does seem unlikely right now. I'll ask around to make sure.

**Comment by Protuhj** [comment]
I'd say supporting this is kinda fundamental to being a back replacement addon.

**Comment by Cidan** [comment]
It's fundamentally impossible to support with custom search -- Blizzard does not expose a way to do it.

How about another custom function, like `BetterBagsSearch("your string")` as a global?

**Comment by Cidan** [comment]
Awesome, I'll add that in soon as I need to do some other work around this anyway :)

**Comment by juntereiner** [comment]
It looks like only adibags use the default search. I've tested it with arkinventory, sorted, baganator, bagnon and none works with /run C_Container.SetItemSearch("soulbound").

**Comment by Cidan** [comment]
> It looks like only adibags use the default search. I've tested it with arkinventory, sorted, baganator, bagnon and none works with /run C_Container.SetItemSearch("soulbound").

Yeah that's right. Anything that implements any sort of improvement to the search engine will bypass the built in search constructs, as Blizzard doesn't allow for any sort of modification to their search logic.
